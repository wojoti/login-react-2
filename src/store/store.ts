/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable @typescript-eslint/default-param-last */
import { createStore } from "redux";

const initialState = {
  value: 0,
};
function counterReducer(state = initialState, action: { type: any }) {
  // Reducers usually look at the type of action that happened
  // to decide how to update the state
  switch (action.type) {
    case "counter/incremented":
      return { ...state, value: state.value + 1 };
    case "counter/decremented":
      return { ...state, value: state.value - 1 };
    default:
      // If the reducer doesn't care about this action type,
      // return the existing state unchanged
      return state;
  }
}
const store = createStore(counterReducer);
