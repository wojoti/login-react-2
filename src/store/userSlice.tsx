/* eslint-disable import/no-cycle */
/* eslint-disable no-param-reassign */
import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import axios from "axios";
import { RootState } from "./store";

export type UserState = {
  success: boolean;
  name: string;
  role: string;
  status: string;
};

export type UserCredentials = {
  username: string;
  password: string;
};

export const initialState: UserState = {
  success: false,
  name: "",
  role: "",
  status: "",
};

export const fetchUserData = createAsyncThunk<UserState, UserCredentials>(
  "user/fetchUserData",
  async ({ username, password }) => {
    const response = await axios
      .get("http://localhost:3010", {
        auth: {
          username,
          password,
        },
      })
      .then((res) => res)
      .catch((err) => err.response);
    return response.data;
  }
);

// type RequestState = "pending" | "fulfilled" | "rejected";

export const userSlice = createSlice({
  name: "user",
  initialState,
  reducers: {
    resetState() {
      // âœ… CORRECT: returns a new value to replace the old one
      return initialState;
    },
    // increment: (state, action) => {
    //   state.value += action.payload;
    // },
  },
  extraReducers: (builder) => {
    // When our request is pending:
    // - store the 'pending' state as the status for the corresponding pokemon name
    builder.addCase(fetchUserData.pending, (state) => {
      state.status = "pending";
    });
    // When our request is fulfilled:
    // - store the 'fulfilled' state as the status for the corresponding pokemon name
    // - and store the received payload as the data for the corresponding pokemon name
    builder.addCase(fetchUserData.fulfilled, (state, action) => {
      state.status = "fulfilled";
      state.name = action.payload.name;
      state.role = action.payload.role;
      state.success = action.payload.success;
    });
    // When our request is rejected:
    // - store the 'rejected' state as the status for the corresponding pokemon name
    builder.addCase(fetchUserData.rejected, (state) => {
      state.status = "rejected";
    });
  },
});

// Action creators are generated for each case reducer function
export const selectStatus = (state: RootState) => state.user.status;
export const selectData = (state: RootState) => state.user;

export const { resetState } = userSlice.actions;
export default userSlice.reducer;
