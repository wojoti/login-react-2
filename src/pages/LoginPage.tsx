import { AppDispatch, RootState } from "@/store/store";
import { fetchUserData } from "@/store/userSlice";
import { LoginTemplate } from "@components/templates";
import { useDispatch, useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";

export interface LoginPageProps {
  testId?: string;
}

function LoginPage({ testId }: LoginPageProps) {
  const navigate = useNavigate();
  const dispatch = useDispatch<AppDispatch>();
  const user = useSelector((state: RootState) => state.user);
  const onLoginSubmit = (
    email: string,
    password: string,
    rememberMe: boolean
  ) => {
    console.log({
      email,
      password,
      rememberMe,
    });
    dispatch(fetchUserData({ username: email, password }));
  };

  const onIconClick = (icon: string) => {
    console.log({ icon });
  };

  const onLinkClick = (path: string) => {
    navigate(path);
  };

  return (
    <LoginTemplate
      onLoginSubmit={onLoginSubmit}
      onIconClick={onIconClick}
      onLinkClick={onLinkClick}
      userData={user}
      testId={testId}
    />
  );
}

export default LoginPage;
