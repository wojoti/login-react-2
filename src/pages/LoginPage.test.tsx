/* eslint-disable @typescript-eslint/no-unused-vars */
import { store } from "@/store/store";
import renderWithProviders from "@/test-utils";
import { fireEvent, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Provider } from "react-redux";
import TestRenderer from "react-test-renderer";
import LoginPage, { LoginPageProps } from "./LoginPage";

const mockedUsedNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

const props: LoginPageProps = {
  testId: "test-logintemplate-id",
};

test("should match snapshot", () => {
  const tree = TestRenderer.create(
    <Provider store={store}>
      <LoginPage {...props} />
    </Provider>
  ).toJSON();
  // const tree = renderWithProviders(<LoginPage {...props} />);
  expect(tree).toMatchSnapshot();
});

test("should render loginpage", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
});

test("should render loginpage default disabled button ", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpageButtonElement = screen.getByTestId("login-button-id");
  expect(loginpageButtonElement).toBeInTheDocument();
  expect(loginpageButtonElement).toBeDisabled();
});

test("should render loginpage with enabled button - both inputs filled, email regex correct", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpageLoginElement = screen.getByTestId("loginform-textinput-login");
  expect(loginpageLoginElement).toBeInTheDocument();
  const textfieldLoginElement = within(loginpageLoginElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldLoginElement).toBeInTheDocument();
  userEvent.type(textfieldLoginElement, "test@test.pl");

  const loginpagePasswordElement = screen.getByTestId(
    "loginform-textinput-password"
  );
  expect(loginpagePasswordElement).toBeInTheDocument();
  const textfieldPasswordElement = within(loginpagePasswordElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldPasswordElement).toBeInTheDocument();
  userEvent.type(textfieldPasswordElement, "test");

  const loginpageButtonElement = screen.getByTestId("login-button-id");
  expect(loginpageButtonElement).toBeInTheDocument();
  expect(loginpageButtonElement).not.toBeDisabled();
});

test("should render loginpage with disabled button - both inputs filled, email regex incorrect", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpageLoginElement = screen.getByTestId("loginform-textinput-login");
  expect(loginpageLoginElement).toBeInTheDocument();
  const textfieldLoginElement = within(loginpageLoginElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldLoginElement).toBeInTheDocument();
  userEvent.type(textfieldLoginElement, "t");

  const loginpagePasswordElement = screen.getByTestId(
    "loginform-textinput-password"
  );
  expect(loginpagePasswordElement).toBeInTheDocument();
  const textfieldPasswordElement = within(loginpagePasswordElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldPasswordElement).toBeInTheDocument();
  userEvent.type(textfieldPasswordElement, "pwd");

  const loginpageButtonElement = screen.getByTestId("login-button-id");
  expect(loginpageButtonElement).toBeInTheDocument();
  expect(loginpageButtonElement).toBeDisabled();
});

test("should render loginpage with disabled button - login filled, email regex correct", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpageLoginElement = screen.getByTestId("loginform-textinput-login");
  expect(loginpageLoginElement).toBeInTheDocument();
  const textfieldLoginElement = within(loginpageLoginElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldLoginElement).toBeInTheDocument();
  userEvent.type(textfieldLoginElement, "test@test.eu");

  const loginpageButtonElement = screen.getByTestId("login-button-id");
  expect(loginpageButtonElement).toBeInTheDocument();
  expect(loginpageButtonElement).toBeDisabled();
});

test("should render loginpage with disabled button - password filled", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpagePasswordElement = screen.getByTestId(
    "loginform-textinput-password"
  );
  expect(loginpagePasswordElement).toBeInTheDocument();
  const textfieldPasswordElement = within(loginpagePasswordElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldPasswordElement).toBeInTheDocument();
  userEvent.type(textfieldPasswordElement, "pwd");

  const loginpageButtonElement = screen.getByTestId("login-button-id");
  expect(loginpageButtonElement).toBeInTheDocument();
  expect(loginpageButtonElement).toBeDisabled();
});

test("should handle click on checkbox", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();

  const loginpageCheckboxAreaElement = screen.getByTestId(
    "loginform-CheckboxArea"
  );
  expect(loginpageCheckboxAreaElement).toBeInTheDocument();

  const checkboxElement = within(loginpageCheckboxAreaElement).getByTestId(
    "checkboxarea-checkbox"
  );
  expect(checkboxElement).not.toBeChecked();
  fireEvent.click(checkboxElement);
  expect(checkboxElement).toBeChecked();
});

test("should render email textfield with red border when invalid (eg. regex)", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpageLoginElement = screen.getByTestId("loginform-textinput-login");
  expect(loginpageLoginElement).toBeInTheDocument();
  const textfieldLoginElement = within(loginpageLoginElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldLoginElement).toBeInTheDocument();
  userEvent.type(textfieldLoginElement, "t");
  expect(textfieldLoginElement).toHaveStyleRule(
    "border",
    "2px solid rgb(244 63 94)"
  );
});

test("should not render password textfield with red border when invalid'", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();
  const loginpageLoginElement = screen.getByTestId(
    "loginform-textinput-password"
  );
  expect(loginpageLoginElement).toBeInTheDocument();
  const textfieldLoginElement = within(loginpageLoginElement).getByTestId(
    "textinput-textfield"
  );
  expect(textfieldLoginElement).toBeInTheDocument();
  userEvent.type(textfieldLoginElement, "t");
  expect(textfieldLoginElement).not.toHaveStyleRule(
    "border",
    "2px solid rgb(244 63 94)"
  );
});
test("should render checkboxarea with checkbox unchecked by default", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();

  const checkboxareaElement = screen.getByTestId("loginform-CheckboxArea");
  expect(checkboxareaElement).toBeInTheDocument();
  const checkboxareaCheckboxElement = within(checkboxareaElement).getByTestId(
    "checkboxarea-checkbox"
  );
  expect(checkboxareaCheckboxElement).toBeInTheDocument();
  expect(checkboxareaCheckboxElement).not.toBeChecked();
});

test("should switch checkbox status after click", () => {
  renderWithProviders(<LoginPage {...props} />);
  const loginpageElement = screen.getByTestId("test-logintemplate-id");
  expect(loginpageElement).toBeInTheDocument();

  const checkboxareaElement = screen.getByTestId("loginform-CheckboxArea");
  expect(checkboxareaElement).toBeInTheDocument();
  const checkboxareaCheckboxElement = within(checkboxareaElement).getByTestId(
    "checkboxarea-checkbox"
  );
  expect(checkboxareaCheckboxElement).toBeInTheDocument();
  fireEvent.click(checkboxareaCheckboxElement);
  expect(checkboxareaCheckboxElement).toBeChecked();
});

test("should render header with text", () => {
  renderWithProviders(<LoginPage {...props} />);
  const headerElement = screen.getByTestId("logintemplate-header");
  expect(headerElement).toBeInTheDocument();
  expect(headerElement).toHaveTextContent("LOGIN");
});

test("should render breakline with text", () => {
  renderWithProviders(<LoginPage {...props} />);
  const breaklineElement = screen.getByTestId("logintemplate-breakline");
  expect(breaklineElement).toBeInTheDocument();
  expect(breaklineElement).toHaveTextContent("OR");
});

test("should render signupform", () => {
  renderWithProviders(<LoginPage {...props} />);
  const signupformElement = screen.getByTestId("logintemplate-socialicons");
  expect(signupformElement).toBeInTheDocument();
});

test("should render matching link with text", () => {
  renderWithProviders(<LoginPage {...props} />);
  const linkElement = screen.getByTestId("logintemplate-link");
  expect(linkElement).toBeInTheDocument();
  expect(linkElement).toHaveTextContent("SIGN UP");
});
