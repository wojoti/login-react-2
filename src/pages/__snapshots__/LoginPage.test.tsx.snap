// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c10 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  height: 32px;
  border-radius: 6px;
  border: 0px solid black;
  background-color: rgb(244 63 94);
  background-color: rgb(244 63 94);
  color: white;
  cursor: pointer;
  box-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
}

.c9 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-right: 10px;
  accent-color: #f43f5e;
  width: 1rem;
  height: 1rem;
}

.c3 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 15px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c11 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c13 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 30px;
  margin-bottom: 30px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 {
  box-sizing: border-box;
  margin: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-shadow: 0 25px 50px -12px rgb(0 0 0 / 0.25);
  padding-top: 2rem;
  padding-bottom: 2rem;
  padding-left: 2.5rem;
  padding-right: 2.5rem;
  background-color: rgb(255 255 255);
  border-style: solid;
  border-width: 0rem;
  border-radius: 0.5rem;
  width: 20rem;
  height: 450px;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 1rem;
  font-weight: 500;
  width: 100%;
}

.c20 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 13px;
  height: 13px;
}

.c5 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: inherit;
  font-size: 14px;
}

.c24 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: rgb(107 114 128);
  font-size: 14px;
}

.c12 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgb(107 114 128);
  font-size: 14px;
  background: none !important;
  border: none;
  padding: 0 !important;
  cursor: pointer;
}

.c25 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: inherit;
  font-size: 14px;
  background: none !important;
  border: none;
  padding: 0 !important;
  cursor: pointer;
}

.c7 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 10px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c23 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  margin-top: 28px;
  margin-bottom: 0px;
  margin-left: 0px;
  margin-right: 0px;
  padding-top: 0px;
  padding-bottom: 0px;
  padding-left: 0px;
  padding-right: 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
  height: 32px;
  border: 2px solid #e5e7eb;
  border-radius: 5px;
  padding-left: 10px;
}

.c14 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 100%;
}

.c15 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  height: 2px;
  width: 100%;
  background-color: rgb(229 231 235);
}

.c16 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  position: relative;
}

.c17 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 1px solid rgb(156 163 175);
  border-radius: 4px;
  position: absolute;
  background-color: white;
  font-size: 14px;
  color: rgb(156 163 175);
  top: 50%;
  left: 50%;
  padding: 1px 2px;
  -webkit-transform: translateX(-50%) translateY(-50%);
  -ms-transform: translateX(-50%) translateY(-50%);
  transform: translateX(-50%) translateY(-50%);
}

.c8 {
  width: 100%;
}

.c19 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: #DE5246;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

.c21 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: #1877f2;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

.c22 {
  box-sizing: border-box;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen", "Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue", sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  border: 2px solid black;
  border-radius: 45px;
  padding: 3px 5px;
  border-color: #0077B5;
  background-color: white;
  margin: 0px 10px;
  cursor: pointer;
}

.c4 {
  width: 100%;
}

.c2 {
  width: 100%;
}

<div
  className="c0"
  data-testid="test-logintemplate-id"
>
  <h1
    className="c1"
    data-testid="logintemplate-header"
  >
    LOGIN
  </h1>
  <div
    className="c2"
    data-testid="logintemplate-loginform"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
        data-testid="loginform-textinput-login"
      >
        <span
          className="c5"
          data-testid="textinput-label"
        >
          Email
        </span>
        <input
          className="c6"
          data-testid="textinput-textfield"
          id="email"
          name="Email"
          onChange={[Function]}
          type="email"
        />
      </div>
    </div>
    <div
      className="c3"
    >
      <div
        className="c4"
        data-testid="loginform-textinput-password"
      >
        <span
          className="c5"
          data-testid="textinput-label"
        >
          Password
        </span>
        <input
          className="c6"
          data-testid="textinput-textfield"
          id="password"
          name="Password"
          onChange={[Function]}
          type="password"
        />
      </div>
    </div>
    <div
      className="c7"
    >
      <div
        className="c8"
        data-testid="loginform-CheckboxArea"
      >
        <input
          className="c9"
          data-testid="checkboxarea-checkbox"
          id="remember"
          name="remember"
          type="checkbox"
        />
        <span
          className="c5"
          data-testid="checkboxarea-label"
        >
          Remember Me?
        </span>
      </div>
    </div>
    <div
      className="c3"
    >
      <button
        className="c10"
        data-testid="login-button-id"
        disabled={false}
        onClick={[Function]}
        type="submit"
      >
        LOGIN
      </button>
    </div>
    <div
      className="c11"
    >
      <button
        className="c12"
        color="rgb(107 114 128)"
        data-testid="loginform-link"
        onClick={[Function]}
        type="button"
      >
        Forgot Password?
      </button>
    </div>
  </div>
  <div
    className="c13"
  >
    <div
      className="c14"
      data-testid="logintemplate-breakline"
    >
      <div
        className="c15"
      >
        <div
          className="c16"
        >
          <p
            className="c17"
          >
            OR
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="c18"
    data-testid="logintemplate-socialicons"
  >
    <button
      className="c19"
      data-testid="iconbutton-g"
      onClick={[Function]}
    >
      <img
        alt="google"
        className="c20"
        data-testid="icon-g"
        height={13}
        src="./img/google.svg"
        width={13}
      />
    </button>
    <button
      className="c21"
      data-testid="iconbutton-fb"
      onClick={[Function]}
    >
      <img
        alt="facebook"
        className="c20"
        data-testid="icon-fb"
        height={13}
        src="./img/facebook.svg"
        width={13}
      />
    </button>
    <button
      className="c22"
      data-testid="iconbutton-l"
      onClick={[Function]}
    >
      <img
        alt="linkedin"
        className="c20"
        data-testid="icon-l"
        height={13}
        src="./img/linkedin.svg"
        width={13}
      />
    </button>
  </div>
  <div
    className="c23"
  >
    <span
      className="c24"
      color="rgb(107 114 128)"
    >
      Need an account?
       
      <button
        className="c25"
        data-testid="logintemplate-link"
        onClick={[Function]}
        type="button"
      >
        SIGN UP
      </button>
    </span>
  </div>
</div>
`;
